eul/Store/MemoryTopStore.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  MemoryTopStore.swift
+//  eul
+//
+//  Created by Jevon Mao on 1/19/21.
+//  Copyright © 2021 Gao Sun. All rights reserved.
+//
+
+import Combine
+import SwiftUI
+import SystemKit
+
+class MemoryTopStore: ObservableObject {
+    private var memorySizeMB = System.physicalMemory() * 1000
+    private var task: Process?
+    private var activeCancellable: AnyCancellable?
+    private var firstLoaded = false
+    @ObservedObject var preferenceStore = SharedStore.preference
+    @Published var dataAvailable = false
+    @Published var topProcesses: [RamUsage] = []
+
+    func update(shouldStart: Bool) {
+        guard shouldStart else {
+            task?.terminate()
+            task = nil
+            return
+        }
+
+        if task != nil {
+            Print("cpu task already started")
+            return
+        }
+        let refreshRate = preferenceStore.smcRefreshRate
+        firstLoaded = false
+        dataAvailable = false
+        topProcesses = []
+        task = shellPipe("top -l 0 -n 5 -stats pid,command,rsize -s \(refreshRate) -orsize") { string in
+            print(string)
+            let rows = string.split(separator: "\n", omittingEmptySubsequences: false).map { String($0) }
+
+            guard let separatorIndex = rows.firstIndex(of: "") else {
+                return
+            }
+
+            if rows.indices.contains(separatorIndex + 1) {
+                let titleRow = rows[separatorIndex + 1].lowercased()
+                if titleRow.contains("pid"), titleRow.contains("mem"), titleRow.contains("command") {
+                    let runningApps = NSWorkspace.shared.runningApplications
+                    let result: [RamUsage] = ((separatorIndex + 2)..<rows.count).compactMap { index in
+                        let row = rows[index].split(separator: " ").map { String($0) }
+                        guard row.count >= 2, let pid = Int(row[0]), let rawRamString = row.last, let ram = Double(rawRamString.filter("0123456789.".contains)) else {
+                            return nil
+                        }
+
+                        let usage = 100 * (ram / self.memorySizeMB)
+
+                        return RamUsage(pid: pid, command: Info.getProcessCommand(pid: pid)!, percentage: usage, usageAmount: ram, runningApp: runningApps.first(where: { $0.processIdentifier == pid }))
+                    }
```

Comment:
maybe we can re-use the top command for better performance. consider merging CPU/mem top stores into one store?